### PR TITLE
feat: add `getContributionsByAddress()`

### PR DESCRIPTION
**Pull request by baumstern**
This PR introduces `getContributionsByAddress(address: Hash)` function, which is essential for the My Actions page. Additionally, This PR added `fetchReportByHCId(hypercertId: string)` function to enable fetching reports by hypercert ID. 

* To demonstrate how `getContributionsByAddress()` can be utilized, modifications have been made to `app/routes/my-actions.tsx`.

### CODE REVIEW

File: app/impact-reports.server.ts [hunk 1]
```diff
@@ -121,6 +121,32 @@ export const fetchReportBySlug = async (slug: string): Promise<Report> => {
...
+			throw new Error(
+				`[server] Report with hypercert Id '${hypercertId}' not found.`,
+			);
+		}
+
+		return foundReport;
+	} catch (error) {
+		console.error(
+			`[server] Failed to get report with hypercert Id '${hypercertId}': ${error}`,
+		);
```
Comment (0xRowdy) [inline app/impact-reports.server.ts:143]:
```suggestion
```

File: app/routes/my-actions.tsx [hunk 2]
```diff
@@ -1,47 +1,52 @@
...
+				description: report.summary,
+			} as HistoryData);
+		}
+	} catch (error) {
+		console.error(
+			`Failed to load reports contributed by ${testAddress}: ${error}`,
+		);
+	}
+	return history;
+};
```
Comment (0xRowdy) [inline app/routes/my-actions.tsx:45]: What happens if one of the reports fails during a loop? Would promise. Do you think promise.all makes sense here?

something like 
```ts
export const loader: LoaderFunction = async () => {
  const testAddress = "0xf39Fd6e51aad88F6F4ce6aB8827279cfffb92266";

  try {
    const contributions = await getContributionsByAddress(testAddress);

    // Map each contribution to a promise that resolves to the history item
    const historyPromises = contributions.map(async (contribution) => {
      const report = await fetchReportByHCId(contribution.hypercert_id);

      return {
        id: contribution.txid,
        date: new Date(contribution.date_created),
        amount: contribution.amount,
        img: {
          src: report.image,
          alt: report.title,
        },
        title: report.title,
        category: report.category,
        location: report.state,
        description: report.summary,
      };
    });

    // Wait for all promises to resolve and assign directly to history
    const history = await Promise.all(historyPromises);

    return history;
  } catch (error) {
    console.error(`Failed to load reports contributed by ${testAddress}: ${error}`);
    // Return an empty array or handle the error as needed
    return [];
  }
};
```

File: app/directus.server.ts [hunk 3]
```diff
@@ -126,6 +133,67 @@ export const getReports = async (): Promise<CMSContent[]> => {
...
+		);
+
+		return validContributions;
+	} catch (error) {
+		console.error(
+			`[Directus] Failed to fetch contributions by address ${address}: ${error}`,
+		);
+		throw new Error(
+			`[Directus] Failed to fetch contributions by address ${address}: ${error}`,
+		);
```
Comment (0xRowdy) [inline app/directus.server.ts:193]: When we throw errors we'll use 
https://ui.shadcn.com/docs/components/sonner
or
https://ui.shadcn.com/docs/components/toast

**Review by 0xRowdy — APPROVED** [review]
Pull in main and Lgtm

File: app/directus.server.ts [hunk 4]
```diff
@@ -126,6 +133,67 @@ export const getReports = async (): Promise<CMSContent[]> => {
...
+				filter: {
+					address: {
+						_eq: getAddress(address),
+					},
+				},
+			}),
+		);
+
+		if (response.length === 0) {
+			console.log(`[Directus] No user found with address: ${address}`);
```
[the remark this answers is not in the document]
Comment (baumstern) [reply app/directus.server.ts:162]: If a user hasn't made any contributions, they wouldn't be registered in the CMS, so I'm thinking this probably isn't an error case, isn't that so?

File: app/impact-reports.server.ts (hunk 1 above)
Comment (baumstern) [reply app/impact-reports.server.ts]: reflected in 66dc3c7

File: app/routes/my-actions.tsx (hunk 2 above)
Comment (baumstern) [reply app/routes/my-actions.tsx]: reflected in 66dc3c7

**Pull request merged**